File: mvpreaderjava/src/main/java/com/example/mvpreaderjava/modle/helper/HttpHelper.java
package com.example.mvpreaderjava.modle.helper;

import com.example.mvpreaderjava.modle.api.WanAndroidApi;
import com.example.mvpreaderjava.modle.bean.JUHE.JUHENewsData;
import com.example.mvpreaderjava.modle.bean.JUHE.JUHENewsResponse;
import com.example.mvpreaderjava.modle.bean.JUHE.JUHENewsResult;

import java.util.List;

import io.reactivex.Flowable;

/**
 * Created by sw116 on 2018/4/19.
 */
public interface HttpHelper extends WanAndroidApi{
    /**
     * 获取聚合新闻列表
     * @param type
     * @return
     */
    Flowable<JUHENewsResponse<JUHENewsResult<List<JUHENewsData>>>> getJuheNews(String type);

}

File: app/src/main/java/com/savion/mvpreader/model/behavior/HttpHelper.kt
package com.savion.mvpreader.model.behavior

import com.savion.mvpreader.model.response.JUHENewsResponse
import io.reactivex.Flowable

/**
 * Created by savion on 2018/4/10.
 */
interface HttpHelper {

    // 获取聚合数据头条信息
    fun getJuheNews(type: String): Flowable<JUHENewsResponse<JUHENewsResponse.JUHENewsResponseResule<List<JUHENewsResponse.JUHENewsResponseData>>>>

}